tests/test_cli.py

```diff
@@ -25,3 +26,40 @@ def test_namespace(namespace, expected, runner):
     assert result.exit_code == 0
 
     assert expected in result.output
+
+
+def test_options_structure(runner):
+    result = runner.invoke(["--options-structure"])
+    assert result.exit_code == 0
+
+    options_json = json.loads(result.output)
+    assert find_conflicts_in_options_dict("snow", options_json) is None
+
+
+def find_conflicts_in_options_dict(path: str, options_dict: Dict[str, Any]):
+
+    keys = list(options_dict.keys())
+    duplicates = {}
+    if "options" in keys:
+        if opts_duplicates := check_options_for_duplicates(options_dict["options"]):
+            duplicates[path] = opts_duplicates
+        keys.remove("options")
+
+    for key in keys:
+        if key_duplicates := find_conflicts_in_options_dict(
+            f"{path} {key}", options_dict[key]
+        ):
+            duplicates.update(key_duplicates)
+
+    if duplicates:
+        return duplicates
+
+    return None
+
+
+def check_options_for_duplicates(options: Dict[str, List[str]]) -> Set[str]:
+    RESERVED_FLAGS = ["--help"]  # noqa: N806
```

Comment:
I removed `-h` reservation because it seems to me that Typer is handling it correctly.
See `--help` for options of the following commands:
```
snow snowpark procedure create --help
snow snowpark procedure describe --help 
```
The first command has own `-h` option and `--help` doesn't duplicate it, while the second command doesn't have own `-h`, so `--help` has a short version.